chrome/background.js

```diff
@@ -1,16 +1,134 @@
-let languages = ['english', 'spanish', 'french', 'portuguese', 'russian']
+let glossaries = {
+	'english': 'english-terminology-for-editors',
+	'spanish': 'spanish-glossary',
+	'french': 'french-glossary',
+	'portuguese': 'portuguese-glossary',
+	'russian': 'russian-glossary'
+}
 
-chrome.runtime.onInstalled.addListener(function() {
+chrome.runtime.onInstalled.addListener(init)
+chrome.runtime.onStartup.addListener(init)
 
-	for (let i in languages) {
-		let lang = languages[i]
-		let glossaryName = lang + '-glossary'
-		const url = chrome.runtime.getURL('data/' + glossaryName + '.json');
+
+function init() {
+	getGlossaries().catch(() => {
+		console.log('Could not get ForumBee. Using local glossary instead.')
+		for (let lang in glossaries) {
+			let glossaryName = lang + '-glossary'
+			const url = chrome.runtime.getURL('data/' + glossaryName + '.json');
+
+			fetch(url).then((resp) => {
+				return resp.json()
+			}).then((data) => {
+				let obj = {}
+				obj[glossaryName] = data
+				chrome.storage.local.set(obj, function() {
+			        console.log(lang + ' glossary set!');
+			    });
+			})
+		}
+	})
+}
+
+async function getGlossaries() {
+	for (let lang in glossaries) {
+		console.log('starting ' + lang)
+		let g = glossaries[lang]
+		let data = []
+		let i = 0
+		let isLast = false
+		while (!isLast) {
+			let url = `https://kiva.forumbee.com/api/2/posts?fields=postKey,title,text&categoryLink=${g}&postType=topics&textFormat=plain&limit=1000&offset=${i*1000}`
```

Comment:
@kevinzwang Assuming 1000 is the max one can fetch at once?

Reply:
That is what it said in the docs